tests/utils/engine_mocking.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+from enum import Enum
+from pathlib import Path
+from typing import Any, List, Type
+from unittest import mock
+
+import numpy
+import onnx
+import onnxruntime as ort
+
+
+class SampleMode(str, Enum):
+    """
+    How to sample outputs generated from the mocked engine.
+    """
+
+    zeros = "zeros"
+    rand = "rand"
+
+
+def mock_engine(*, rng_seed: int, mode: SampleMode = SampleMode.rand, **kwargs):
+    """
+    Intended to create a fake engine instead of compiling the model.
+
+    This patches the `LIB.deepsparse_engine()` that is used in deepsparse.Engine
+    with `_FakeDeepsparseLibEngine`.
```

Comment:
Python is an amazing evil

Reply:
😁 I like it for unit testing since it lets you isolate parts of the code that you want the test to focus on even if the code's api doesn't allow for that. Of course I also think code should be written so that its easy to test without doing stuff like this, but that's not always feasible.